qemu/tests/uefi_check_descriptor_meta-files.py

```diff
@@ -0,0 +1,69 @@
+import re
+import json
+import logging
+
+from avocado.utils import process
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Verify descriptor meta-files after ovmf package installation on the host:
+    1) Check edk2-ovmf package has been installed already.
+    2) Qurey file list on edk2-ovmf package.
+    3) Make sure the descriptor meta-files will be in file list.
+       2 files in total.
+       For rhel7: 50-ovmf-sb.json, 60-ovmf.json
+       For rhel8: 40-edk2-ovmf-sb.json, 50-edk2-ovmf.json
+    4) Check the JSON files internally.
+       check that the "filename" elements in both files point to valid files.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def check_element(filename, file_list):
+        """
+        check 'filename' element point to a valid file
+
+        :param filename: 'filename' element
+        :param file_list: query files command output
+        """
+        if filename not in file_list:
+            test.fail("The 'filename' element in meta-file point to an "
+                      "invalid file. The invalid file is '%s'" % filename)
+
+    query_package = params["query_package"]
+    error_context.context("Check edk2-ovmf package has been "
+                          "installed already", logging.info)
+    status, output = process.getstatusoutput(query_package,
+                                             ignore_status=True,
+                                             shell=True)
+    if status:
+        test.error("Please install edk2-ovmf package on host.")
+    package_name = params["ovmf_package_name"]
+    ovmf_package = re.findall(package_name, output, re.S)
+    if not ovmf_package:
+        test.error("Not found right edk2-ovmf package on host. "
+                   "The actual output is '%s'" % output)
+    query_files = params["query_files"] % ovmf_package[0]
+    file_suffix = params["file_suffix"]
+    meta_files = []
+    output = process.getoutput(query_files, shell=True)
+    for line in output.splitlines():
+        if line.endswith(file_suffix):
+            meta_files.append(line)
+    if len(meta_files) != 2:
```

Comment:
any possibility that more json files will be added in future? I check my fedora system, 3 json files are found.
so can we use an option defined in cfg instead of hard code 2 here?

Reply:
As far as I know, the number of files are always 2 on rhel host. I will define it in cfg. thanks